### PR TITLE
🐛 Fix upcoming events predicate for viewers west of UTC

### PR DESCRIPTION
**Pull request by lukevella**
## Summary

Fixes RAL-1256

Two divergent copies of the "upcoming" predicate existed, and both were wrong differently:

- `trpc/routers/dashboard.ts` compared the instant of local midnight (`dayjs().tz(user.timeZone).startOf("day")`) against allDay starts stored at UTC midnight, so for any viewer west of UTC, today's all-day events never counted as upcoming.
- `features/scheduled-event/data.ts` (`getUpcomingEvents`/`getPastEvents`) used midnight in the server process zone, which is viewer-independent and only correct on a UTC runner.

Because the two differed, the dashboard count and the events list could disagree for the same user.

Root cause: for `allDay` rows, `start`/`end` are not instants — they are calendar dates encoded as UTC midnight. Date columns must be compared date-to-date.

## Fix

- **Date helpers** in `lib/datetime/utils.ts`: `getCalendarDate(now, timeZone)` (pure Intl, en-CA → YYYY-MM-DD) and `calendarDateToUTCMidnight(date)`, which encode "today in the viewer's zone" the same way allDay dates are stored.
- **One shared predicate** in `features/scheduled-event/predicates.ts`: `upcomingScheduledEventWhere({ now, timeZone })`, used by dashboard stats, the upcoming list, and (as its exact negation, `pastScheduledEventWhere`) the past list, so past/upcoming partition with no event in neither or both. `end` is stored exclusive (start + 1 day), so `end > todayUtcMidnight` keeps an event through its final day and handles multi-day spans. The timed arm uses `end > now` so in-progress meetings still count as upcoming.
- **Viewer zone as input**: the client passes the browser zone (validated by a new `timeZoneSchema` in `lib/datetime/schema.ts`) into `dashboard.stats` and `events.infiniteList`.
- **Index**: added `@@index([spaceId, status, start])` on `ScheduledEvent` with a generated migration.

## Viewer-zone fallback decision

The `timeZone` input is optional. Server-side resolution is `input.timeZone ?? normalizeTimeZone(ctx.user.timeZone) ?? "UTC"` — the stored user zone, and only then UTC. The SSR prefetches pass the server's own zone so the suspense query on the dashboard hits the prefetched cache instead of fetching (cookie-less) during SSR; when the browser zone differs from the server zone, the client refetches with the real browser zone after hydration. The RAL-1245 timezone cookie work will restore full prefetch parity.

## Deliberately deferred

The "Guard the invariant" section of the ticket (DB CHECK constraints, backfill, and the polls.ts finalize-path change) is intentionally out of scope for this PR.

## Test coverage

Vitest unit tests (`pnpm test:unit`), all passing:

- `getCalendarDate` west of UTC (Honolulu), east of UTC (Tokyo), and UTC; `calendarDateToUTCMidnight` encoding
- Honolulu scenario from the ticket: viewer at 3pm July 1 Honolulu sees an all-day event stored `Jul 1 00:00Z` as upcoming (and a UTC viewer at the same instant sees it as past)
- Multi-day all-day span stays upcoming through its final day
- In-progress timed event counts as upcoming
- Past is the exact negation of upcoming across three zones and boundary cases (all-day ending exactly at today's UTC midnight, timed ending exactly at now)

`pnpm check`, `pnpm type-check`, and `pnpm test:unit` all pass.

🤖 Generated with [Claude Code](https://claude.com/claude-code)

## Summary by CodeRabbit

* **New Features**
  * Event and dashboard views now respect your browser time zone for more accurate dates and upcoming/past event listings.
  * Scheduled event handling has been improved to better match all-day and timed events across time zones.

* **Bug Fixes**
  * Fixed inconsistencies in upcoming and past event results when viewing from different time zones.

* **Performance**
  * Added database indexing to help event queries load faster.

### CODE REVIEW

File: apps/web/src/features/scheduled-event/predicates.ts [hunk 1]
```diff
@@ -0,0 +1,53 @@
...
+  );
+  return {
+    status: "confirmed",
+    deletedAt: null,
+    OR: [
+      { allDay: false, end: { lte: now } },
+      { allDay: true, end: { lte: todayUtcMidnight } },
+    ],
+  } satisfies Prisma.ScheduledEventWhereInput;
+}
```
[the remark this answers is not in the document]
Comment (lukevella) [reply apps/web/src/features/scheduled-event/predicates.ts]: Addressed in 879240d52 with a stronger variant: both predicates now delegate to a single private `scheduledEventWhere({ now, timeZone, past })` core, so the boundary is computed once and the gt/lte complements sit side by side — the exact-negation guarantee is structural rather than conventional. Public API unchanged; negation tests still pass.

**Comment by lukevella** [comment]
@coderabbitai review
